server/src/main/java/org/elasticsearch/search/aggregations/metrics/ValueCountAggregator.java

```diff
@@ -66,6 +66,34 @@ public LeafBucketCollector getLeafCollector(LeafReaderContext ctx,
             return LeafBucketCollector.NO_OP_COLLECTOR;
         }
         final BigArrays bigArrays = context.bigArrays();
+
+        if (valuesSource instanceof ValuesSource.Numeric) {
```

Comment:
I think it'd be nice to have a "value counting" member in `ValuesSource` but I'd be happy to merge this as is and open up a follow up myself to do that. Or you can, if you want @xjtushilei.

Reply:
As much as it is annoying to put something just for one agg in `ValuesSource`, I think it'd be nice to have a something in there just so it is more obvious that we do these sorts of things to the values.

Another option is to use the values source refactor to plug in different count implementations. That'd probably be more in line with the direction we're going now.

Reply:
Either way, I'm happy to take this as is and do the twisting around myself in a follow up change and ping you for review.

Reply:
> I think it'd be nice to have a "value counting" member in `ValuesSource` but I'd be happy to merge this as is and open up a follow up myself to do that. Or you can, if you want @xjtushilei.

yes, you can do it.